tests/reference/c-test_builtin_str-7e198f7.stdout

```diff
@@ -0,0 +1,139 @@
+            printf("%s%s", "ASSERT failed: ", __FILE__);                       \
+            printf("%s%s", "\nfunction ", __func__);                           \
+            printf("%s%d%s", "(), line number ", __LINE__, " at \n");          \
+            printf("%s%s", #cond, "\n");                                       \
+            printf("%s", "ERROR MESSAGE:\n");                                  \
+            printf("%s%s", msg, "\n");                                         \
+            exit(1);                                                           \
+        }                                                                      \
+    }
+
+
+struct dimension_descriptor
+{
+    int32_t lower_bound, length;
+};
+// Forward declarations
+void _lpython_main_program();
+void str_conv_for_variables();
+void test_issue_883();
+void test_str_int();
+void test_str_slice_step();
+
+// Implementations
+void _lpython_main_program()
+{
+    test_str_int();
+    str_conv_for_variables();
+    test_str_slice_step();
+    test_issue_883();
+}
+
+void str_conv_for_variables()
+{
+    bool bool_t;
+    char * str_t;
+    int32_t x;
+    float xx;
+    double yy;
+    x = 123;
+    ASSERT(strcmp("123", _lfortran_int_to_str4(x)) == 0);
+    x = 12345;
+    ASSERT(strcmp("12345", _lfortran_int_to_str4(x)) == 0);
+    x = -12;
+    ASSERT(strcmp("-12", _lfortran_int_to_str4(x)) == 0);
+    x = -121212;
+    ASSERT(strcmp("-121212", _lfortran_int_to_str4(x)) == 0);
+    xx =   1.23222339999999999e+01;
+    ASSERT(strcmp(_lfortran_float_to_str4(xx), "12.322234") == 0);
+    yy =   1.23222339999999999e+01;
+    ASSERT(strcmp(_lfortran_float_to_str8(yy), "12.322234") == 0);
+    bool_t = true;
+    ASSERT(strcmp(bool_t ? "True" : "False", "True") == 0);
+    bool_t = false;
+    ASSERT(strcmp(bool_t ? "True" : "False", "False") == 0);
+    str_t = "just a str";
+    ASSERT(strcmp(str_t, str_t) == 0);
+}
+
+void test_issue_883()
+{
+    int32_t __explicit_iterator;
+    char * __tmp_assign_for_loop;
+    char * c;
+    char * s;
+    s = "abcde";
+    for (__explicit_iterator=0; __explicit_iterator<=strlen(__tmp_assign_for_loop) - 1; __explicit_iterator++) {
+        c = (char *)memcpy((char *)calloc(2U, sizeof(char)), __tmp_assign_for_loop + __explicit_iterator + 1 - 1, 1U);
+        printf("%s\n", c);
+    }
+}
+
+void test_str_int()
+{
+    char * s;
+    s = _lfortran_int_to_str4(356);
+    ASSERT(strcmp(s, "356") == 0);
+    s = _lfortran_int_to_str4(-567);
+    ASSERT(strcmp(s, "-567") == 0);
+    ASSERT(strcmp(_lfortran_int_to_str4(4), "4") == 0);
+    ASSERT(strcmp(_lfortran_int_to_str4(-5), "-5") == 0);
+    ASSERT(strcmp("", "") == 0);
+    ASSERT(strcmp("1234", "1234") == 0);
+    ASSERT(strcmp(false ? "True" : "False", "False") == 0);
+    ASSERT(strcmp(true ? "True" : "False", "True") == 0);
+    ASSERT(strcmp("just a str", "just a str") == 0);
+    ASSERT(strcmp(_lfortran_float_to_str8(  1.21234000000000002e+01), "12.1234") == 0);
```

Comment:
The only failing test is this. The error comes from this line, `_lfortran_float_to_str8(  1.21234000000000002e+01)` gives: `"12.123400"` but RHS is `"12.1234"`. I think we have the same issue with LLVM in precision but it was not raised by LLVM backend because it visits the `x.m_value` which is not null in this case and so doesn't raise an error there.

Reply:
The original test is `assert str(12.1234) == "12.1234"` which is wrong. You can't test floating point numbers like that. You could (in this case) do it like this: `assert str(12.1234)[:7] == "12.1234"` and hopefully it would work.